### PR TITLE
[BUG] deprecate `fitted_list` attribute of `_HeterogenousEnsembleForecaster` and 

### PR DESCRIPTION
**Pull request by sanskarmodi8**
I tried solving issue #7418 by slightly modifying the `HierarchyEnsembleForecaster` code.

### CODE REVIEW

**Comment by sanskarmodi8** [comment]
Hi,
After further exploration, I believe this enhancement needs more foundational changes than initially expected. I think I’ll close this PR for now. Sorry for the inconvenience caused!

**Comment by felipeangelimvieira** [comment]
Hi @sanskarmodi8! Could you explain what you mean by foundational changes?

**Comment by sanskarmodi8** [comment]
Hey @felipeangelimvieira ,
Apologies for any confusion caused! By "foundational changes," I meant that the proposed solution was failing several test cases, and I believed it might require deeper modifications to ensure compatibility.

**Comment by sanskarmodi8** [comment]
The issue #7418 was all about the attribute `forecasters_` not containing fitted forecasters, instead `fitted_list` was doing this, which was unconventional.

So we can just directly fit the forecasters present in `self.forecasters_`, instead of training a clone and adding it to `fitted_list`. This will make `self.forecasters_` to hold the fitted forecasters.
And we can display a deprecation warning for `self.fitted_list` instead of completely removing it right now. It should work the same as in the original code.

Below is my proposed solution about how can we handle this issue. Let me know if I'm headed in the wrong direction.

``` python
@property
def fitted_list(self):
    """Deprecated attribute."""

    import warnings
    warnings.warn(
        """The `fitted_list` property is deprecated
        and will be removed in future versions.
        Please use `forecasters_` instead.
        """,
        DeprecationWarning,
    )
    return [
        [forecaster, indices]
        for (_, forecaster, level), indices in zip(self.forecasters_, self._indices_)
    ]

def _fit(self, y, X, fh):
    """Fit to training data.

    Parameters
    ----------
    y : pd-multiindex
        Target time series to which to fit the forecaster.
    fh : int, list or np.array, optional (default=None)
        The forecasters horizon with the steps ahead to predict.
    X : pd.DataFrame, optional (default=None)
        Exogenous variables are ignored.

    Returns
    -------
    self : returns an instance of self.
    """
    # Creating aggregated levels in data
    if _check_index_no_total(y):
        z = self._aggregate(y)
    else:
        z = y

    if X is not None:
        if _check_index_no_total(X):
            X = self._aggregate(X)

    x = X

    # check forecasters
    self.forecasters_ = self._check_forecasters(y, z)
    self._indices_ = []

    if y.index.nlevels == 1:
        self.forecasters_[0][1].fit(y, fh=fh, X=X)
        self._indices_.append(y.index)
        return self

    if self.by == "level":
        hier_dict = self._get_hier_dict(z)
        for _, forecaster, level in self.forecasters_:
            if level in hier_dict.keys():
                df = z[z.index.droplevel(-1).isin(hier_dict[level])]
                if X is not None:
                    x = X.loc[df.index]
                forecaster.fit(df, fh=fh, X=x)
                self._indices_.append(df.index.droplevel(-1).unique())

    else:
        node_dict, frcstr_dict = self._get_node_dict(z)
        for key, nodes in node_dict.items():
            df = z[z.index.droplevel(-1).isin(nodes)]
            if X is not None:
                x = X.loc[df.index]
            frcstr_dict[key].fit(df, fh=fh, X=x)
            self._indices_.append(df.index.droplevel(-1).unique())
    return self
```

**Comment by fkiraly** [comment]
This looks great - I think this is already what you did above?

The version to deprecate would be 0.37.0, given the deprecation policy.
Please add a `todo 0.37.0` comment as well, to remove this.

**Comment by sanskarmodi8** [comment]
Hey @fkiraly and @felipeangelimvieira,

I have updated this PR to potentially solve the issue #7418.
Waiting for your feedback on this solution.

Thanks!

File: sktime/forecasting/compose/_hierarchy_ensemble.py [hunk 1]
```diff
@@ -159,6 +159,23 @@ def _aggregate(self, y):
...
+    @property
+    def fitted_list(self):
+        """Deprecated attribute.
+
+        TODO 0.37.0: Remove this property entirely.
+        """
+        import warnings
+
+        warnings.warn(
+            """The fitted_list property is deprecated
```
Comment (fkiraly) [inline sktime/forecasting/compose/_hierarchy_ensemble.py:171]: Kindly mention the name of the forecaster, so the user knows which component it is coming from

File: sktime/forecasting/compose/_hierarchy_ensemble.py [hunk 2]
```diff
@@ -159,6 +159,23 @@ def _aggregate(self, y):
...
 
+    @property
+    def fitted_list(self):
+        """Deprecated attribute.
+
+        TODO 0.37.0: Remove this property entirely.
+        """
+        import warnings
+
+        warnings.warn(
```
Comment (fkiraly) [inline sktime/forecasting/compose/_hierarchy_ensemble.py:170]: this should use `sktime.utils.warnings.warn`, so users can turn it off via `set_config`.

**Review by fkiraly — CHANGES_REQUESTED** [review]
Great, looks good - thanks!

I merged it with my approach here: https://github.com/sktime/sktime/pull/7522 - which moves the cloning into a separate method, to make reading the code easier.

Could you kindly have a look whether it still makes sense?

You can also add yourself as an author to the estimator, given how substantial your fix is.

**Comment by sanskarmodi8** [comment]
> Great, looks good - thanks!
> 
> I merged it with my approach here: #7522 - which moves the cloning into a separate method, to make reading the code easier.
> 
> Could you kindly have a look whether it still makes sense?
> 
> You can also add yourself as an author to the estimator, given how substantial your fix is.

Thank you for reviewing and refining the changes, and for integrating your approach to improve readability. The updated implementation looks great and solves the issue effectively.

Regarding adding myself as an author, I appreciate your encouragement. Could you confirm if it would still be appropriate given that the final implementation incorporates your changes as well?

**Review by fkiraly — DISMISSED** [review]
Sure it is appropriate - especially since your edits contributed to informing my changes.

PS: I reverted some renaming changes that do not change logic.

**Pull request merged**